### PR TITLE
reader should have a shards dir

### PR DESCRIPTION
**Pull request by hermeGarcia**
### Description
Describe the proposed changes made in this PR.

### How was this PR tested?
Describe how you tested this PR.

### CODE REVIEW

**Review by javitonino — APPROVED** [review]

File: nucliadb_node/src/grpc/collect_garbage.rs [hunk 1]
```diff
@@ -43,9 +43,12 @@ pub async fn garbage_collection_loop(parameters: GCParameters, cache: Arc<ShardW
     loop {
         tokio::time::sleep(parameters.loop_interval).await;
 
-        let Ok(shards_dir_iterator) = read_dir(&parameters.shards_path) else {
-            error!("Garbage collection loop can not read shards directory");
-            break;
+        let shards_dir_iterator = match read_dir(&parameters.shards_path) {
+            Ok(iterator) => iterator,
+            Err(error) => {
+                error!("Update loop can not read shards directory: {error:?}");
```
Comment (jotare) [inline nucliadb_node/src/grpc/collect_garbage.rs:49]:
```suggestion
                error!("Garbage collector loop can not read shards directory: {error:?}");
```

File: nucliadb_node/src/grpc/update.rs [hunk 2]
```diff
@@ -43,9 +43,12 @@ pub async fn update_loop(parameters: UpdateParameters, cache: Arc<ShardReaderCac
...
         tokio::time::sleep(parameters.refresh_rate).await;
 
-        let Ok(shards_dir_iterator) = read_dir(&parameters.shards_path) else {
-            error!("Update loop can not read shards directory");
-            break;
+        let shards_dir_iterator = match read_dir(&parameters.shards_path) {
+            Ok(iterator) => iterator,
+            Err(error) => {
+                error!("Update loop can not read shards directory: {error:?}");
+                continue;
```
Comment (jotare) [inline nucliadb_node/src/grpc/update.rs:50]: Maybe some sleep would be nice to not log in a loop until writer starts

File: nucliadb_node/src/grpc/collect_garbage.rs [hunk 3]
```diff
@@ -43,9 +43,12 @@ pub async fn garbage_collection_loop(parameters: GCParameters, cache: Arc<ShardW
...
+                continue;
```
Comment (jotare) [inline nucliadb_node/src/grpc/collect_garbage.rs:50]: Same sleep idea here

File: nucliadb_node/src/grpc/update.rs (hunk 2 above)
Comment (hermeGarcia) [reply nucliadb_node/src/grpc/update.rs:50]: Its in a loop, there is an sleep at the start of it

**Pull request merged**